### PR TITLE
chore: add migration guides templates

### PR DESCRIPTION
**Pull request by boliveira**
## Description

- This adds migration guides templates for all packages to document all
the necessary changes for the migration to the new version.

- Fixed the names of test files `getPreferences` and `putPreferences`
to `getUserPreferences` and `putUserPreferences` respectively.

- Added missing exports to action factories.

- Minor internal repository fixes.

## Checklist

- [X] The commit message follows our guidelines
- [X] Tests for the respective changes have been added
- [X] The code is commented, particularly in hard-to-understand areas
- [X] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by ruifcnunes — APPROVED** [review]

**Review by edmundofreitas — APPROVED** [review]

**Comment by nelsonleite** [comment]
Please review the table of contents of the `packages/redux/MIGRATION.md`, its indentation doesn't seem right 🤔 (almost everything is at the same level, and I think at lease the "What's new" and "What you need to change" should be indented under the respective section)

**Review by nelsonleite — COMMENTED** [review]
Please review the table of contents of the `packages/redux/MIGRATION.md`, its indentation doesn't seem right 🤔 (almost everything is at the same level, and I think at lease the "What's new" and "What you need to change" should be indented under the respective section)

File: packages/client/MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+  - [Addresses](#addresses)
+  - [Authentication](#authentication)
+  - [Categories](#categories)
+  - [Checkout](#checkout)
+  - [Contents](#contents)
+  - [Designers](#designers)
+  - [Helpers](#helpers)
+  - [Locale](#locale)
+  - [Management](#management)
+  - [OAuth](#oAuth)
```
Comment (nelsonleite) [inline packages/client/MIGRATION.md:17]:
```suggestion
  - [OAuth](#oauth)
```

File: packages/react/MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
+This guide will help you migrate `@farfetch/blackout-react` to the latest version.
+
+# Table of Contents
+
+- [Migrating from react v0 to v1](#migrating-from-react-v0-to-v1)
+
+  - [Analytics](#analytics)
+  - [Bags](#bags)
```
Comment (nelsonleite) [inline packages/react/MIGRATION.md:8]: Not in use, you can delete it

File: packages/redux/MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,2166 @@
...
+  - [Entities](#entities)
+  - [Forms](#forms)
+  - [Helpers](#helpers)
+  - [Locale](#locale)
+  - [Loyalty](#loyalty)
+  - [Management](#management)
+  - [Merchants Locations](#merchants-locations)
+  - [Orders](#orders)
+  - [Payments](#payments)
+  - [Products/Details](#products/details)
```
Comment (nelsonleite) [inline packages/redux/MIGRATION.md:25]:
```suggestion
  - [Products/Details](#productsdetails)
```

File: packages/redux/MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,2166 @@
...
+  - [Products/Listing](#products/listing)
```
Comment (nelsonleite) [inline packages/redux/MIGRATION.md:26]:
```suggestion
  - [Products/Listing](#productslisting)
```

**Review by nelsonleite — CHANGES_REQUESTED** [review]

File: packages/client/MIGRATION.md (hunk 1 above)
Comment (boliveira) [reply packages/client/MIGRATION.md]: Fixed

File: packages/redux/MIGRATION.md (hunk 3 above)
Comment (boliveira) [reply packages/redux/MIGRATION.md]: I used VSCode's to test the links and they worked there. Fixed, thanks!

File: packages/redux/MIGRATION.md (hunk 4 above)
Comment (boliveira) [reply packages/redux/MIGRATION.md]: Fixed, thanks!

**Review by nelsonleite — APPROVED** [review]

**Pull request merged**